source/common/config/utility.cc

```diff
@@ -159,37 +159,64 @@ std::string Utility::resourceName(const ProtobufWkt::Any& resource) {
       fmt::format("Unknown type URL {} in DiscoveryResponse", resource.type_url()));
 }
 
-std::vector<Stats::TagExtractorPtr>
-Utility::createTagExtractors(const envoy::api::v2::Bootstrap& bootstrap) {
-  std::vector<Stats::TagExtractorPtr> tag_extractors;
-
-  // Ensure no tag names are repeated.
+void Utility::detectTagNameConflict(const envoy::api::v2::Bootstrap& bootstrap) {
```

Comment:
Not sure what others think about this, but is there a way we could get the name conflict detection as a part of the creation of TagExtractors?  We could do that in a few ways: 
1.  We could create a stateful class for doing this initial config processing, keeping track of `TagExtractors`, and taking care of doing stat creation-time processing (like `getTagsForName()` in `ThreadLocalStore`).  This is probably the elegant and more maintainable solution because it has the benefit of reducing complexity in other classes, too.
2.  We could pass mutable state between these utility methods.  This would be a bit ugly, but it would get rid of the nearly repeated code.

Reply:
Agree. I was also thiking how we generalize tag extractors and default tags. It's off-topic but I started with creating tag extrators which do match every stat names but it was not good idea.

(1) sounds good to me, so I'll try (1). Please tell me the possible name of the class if you have ideas. The class will consume bootstrap configurations and build it-self, then create proper tags when a stats name will be given. Maybe `TagProvider` or `TagProducer`?

Reply:
Yeah, I thought about the idea of a generalized tag extractor as well, and for [performance reasons](https://github.com/envoyproxy/envoy/issues/2373), I agree with you that it's probably best to not do it that way.

Great!  Yeah, I'm not too great with naming, but both of those seem reasonable to me.

Reply:
Implemented (1) in e0596988a490ae5a0ad78271523d2cc9046d6896. Please note I didn't move bootrap config consuming task from Config::Utility bacause (a). the TagProducer (and its factory) living in the stats component shouldn't depend on Bootrap config (b) the consuming task is not so big from the both points of code size and responsibility. If you have a better place for moving the task, I'm glad to implement that.